ksqldb-engine/src/main/java/io/confluent/ksql/planner/LogicalPlanner.java

```diff
@@ -799,6 +811,27 @@ private boolean isInnerNode(final PlanNode node) {
     throw new IllegalStateException("Unknown node type: " + node.getClass().getName());
   }
 
+  private JoinKey buildForeignJoinKey(final Join join, final ColumnReferenceExp columnRef) {
+    final AliasedDataSource leftSource = join.getInfo().getLeftSource();
+    final SourceName alias = leftSource.getAlias();
+    final List<QualifiedColumnReferenceExp> leftSourceKeys =
+        leftSource.getDataSource().getSchema().key().stream()
+            .map(c -> new QualifiedColumnReferenceExp(alias, c.name()))
+            .collect(Collectors.toList());
+
+    final QualifiedColumnReferenceExp keyColumn = leftSourceKeys.get(0);
+
+    final ColumnName keyColumnName = ColumnNames
+        .generatedJoinColumnAlias(keyColumn.getQualifier(), keyColumn.getColumnName());
+
+    final ColumnName foreignKeyColumnName = columnRef.maybeQualifier().isPresent()
+        ? ColumnNames.generatedJoinColumnAlias(
```

Comment:
Why are we rewriting the column name here? This column name needs to be a column in the left source's schema, else the physical plan generation will fail.

Reply:
While debugging, I found the `columnRef.getColumnName()` did not contain the column alias. I had to generate the column again using the qualifier + column name. But now I think about it, if we know this FK column is from the left source, then there is no need to specify the alias, right? Is that what you mean that it is not necessary?

Reply:
Not only is it unnecessary, I think it is also incorrect. `generatedJoinColumnAlias()` actually changes the column name to append the qualifier in front of the column name. If the column name is `FOO` and the qualifier is `Q`, then this code means we'll be looking for `Q_FOO` in the left source schema, which does not exist. Only `FOO` exists.

Reply:
I tried to address all open question in #7543 expect this one. Not 100% sure if I understand the scope yet.

A simple rewrite did not work, as we use `QualifiedColumnReferenceExp` while `leftSchema.key()` returns a list of `Column` object... Not sure how to translated between both and why we even use `QualifiedColumnReferenceExp` to begin wtih.